### PR TITLE
Import existing CVAT datasets into FiftyOne (#2)

### PR DESCRIPTION
**Pull request by brimoor**
An alternative implementation of https://github.com/voxel51/fiftyone/pull/1456 that also implements the TODO items mentioned there.

Other changes
- Added new ``unexpected="keep"`` option to `load_annotations()`
- `%06d_` is no longer prepended to video filenames in CVAT, since it is not necessary (always one video per task)
- Optimized database I/O in `_create_shapes_tags_tracks()` and `_build_frame_id_map()`

## Image dataset example

```py
import os

import fiftyone as fo
import fiftyone.core.utils as fou
import fiftyone.zoo as foz
import fiftyone.utils.cvat as fouc

dataset = foz.load_zoo_dataset("quickstart")
dataset = dataset.select_fields("ground_truth").clone()

# Create a CVAT project with labels
task_size = 10
results = dataset.annotate(
    "test",
    label_field="ground_truth",
    task_size=task_size,
    segment_size=3,
    project_name="test",
)
# print(results.task_ids)  # if using `task_ids` instead of `project_name`

#
# In order to load annotations, we need a mapping from CVAT image filenames to
# filepaths on local disk
#
# Since we're using a CVAT task uploaded via FiftyOne, the mapping is a bit
# weird
#
data_map = {}
for batch in fou.iter_batches(dataset.values("filepath"), task_size):
    for idx, filepath in enumerate(batch):
        filename = "%06d_%s" % (idx, os.path.basename(filepath))
        data_map[filename] = filepath

# Default: import into fields matching the label type (eg "detections")
fouc.import_annotations(
    dataset,
    project_name="test",
    data_path=data_map,
)

# But, we can instead prompt for field names
fouc.import_annotations(
    dataset,
    project_name="test",
    data_path=data_map,
    label_types="prompt",
)

# Or we can explicitly specify where to import
fouc.import_annotations(
    dataset,
    project_name="test",
    data_path=data_map,
    label_types={"detections": "ground_truth2"},
)

session = fo.launch_app(dataset)
```

## Video dataset example

```py
import os

import fiftyone as fo
import fiftyone.core.utils as fou
import fiftyone.zoo as foz
import fiftyone.utils.cvat as fouc

dataset = foz.load_zoo_dataset("quickstart-video").clone()

# Create a CVAT project with labels
results = dataset.annotate(
    "test",
    label_field="frames.detections",
    project_name="test-video",
)

dataset.rename_frame_field("detections", "detections_orig")

fouc.import_annotations(dataset, project_name="test-video")

print(dataset.count_values("frames.detections_orig.detections.label"))
print(dataset.count_values("frames.detections.detections.label"))
```

### CODE REVIEW

**Review by ehofesmann — APPROVED** [review]
LGTM! There are two issues that I resolved in #1513 (`data_path` points to a json file and that `build_frame_id_map` needed to ensure frames). We may want to merge that in before merging this into develop.

One thing to note is that different tasks in CVAT can use the same filename for different images. This caused some weird behavior, but after thinking about it, I decided that we should just continue to treat them as if they were the same image and leave it to the user to ensure unique filenames in CVAT. We may want to think about logging a warning about it though.

**Comment by brimoor** [comment]
> LGTM! There are two issues that I resolved in #1513 (`data_path` points to a json file and that `build_frame_id_map` needed to ensure frames). We may want to merge that in before merging this into develop.
> 
> One thing to note is that different tasks in CVAT can use the same filename for different images. This caused some weird behavior, but after thinking about it, I decided that we should just continue to treat them as if they were the same image and leave it to the user to ensure unique filenames in CVAT. We may want to think about logging a warning about it though.

Thanks for the review 💪 Your comments made me realize there was previously a bug in situations where a user provides a `data_path` that only includes a subset of the files in the specified tasks. I updated things so that it correctly handles that case now, which required adding the same fixes you mentioned in #1513.

**Pull request merged**
